### PR TITLE
apply showGridLines axis option

### LINKED ISSUE #353: apply showGridLines axis option

### PR DESCRIPTION
**Pull request by wflemingnz-retired**
closes #353

### CODE REVIEW

File: projects/ngx-d3timeline/src/lib/axis/axis-component.ts [hunk 1]
```diff
@@ -10,12 +10,14 @@ import { Axis } from './axis';
         ngx-d3timeline-axis-tick-mark
         [tickMark]="tickMark"
       ></svg:g>
-      <svg:g
-        *ngFor="let gridLine of axis.gridLines"
-        class="grid-line"
-        ngx-d3timeline-line
-        [line]="gridLine"
-      ></svg:g>
+      <svg:g *ngIf="axis.showGridLines">
```
Comment (umarmohammed) [inline projects/ngx-d3timeline/src/lib/axis/axis-component.ts]: maybe use ng-container here?

File: projects/ngx-d3timeline/src/lib/axis/axis.ts [hunk 2]
```diff
@@ -4,5 +4,6 @@ import { Line } from '../core/line';
 export interface Axis {
   tickMarks: TickMark[];
   line: Line;
+  showGridLines: boolean;
```
Comment (wflemingnz-retired) [inline projects/ngx-d3timeline/src/lib/axis/axis.ts:7]: I considered leaving this off but not generating the grid lines further up if it was false. Currently its a bit inefficient but maybe more definitive. What do you think?

File: projects/ngx-d3timeline/src/lib/options/options.selectors.ts [hunk 3]
```diff
@@ -26,3 +27,16 @@ export const selectResourceAxisOptions = createSelector(
   selectOptions,
   options => options.resourceAxis
 );
+
+const getShowGridLines = (axisOptions: AxisOptions) =>
```
Comment (umarmohammed) [inline projects/ngx-d3timeline/src/lib/options/options.selectors.ts]: maybe put this as a named function in options-utils to keep consistent with how its done elsewhere

**Review by umarmohammed — CHANGES_REQUESTED** [review]

File: projects/ngx-d3timeline/src/lib/axis/axis-component.ts (hunk 1 above)
Comment (wflemingnz-retired) [reply projects/ngx-d3timeline/src/lib/axis/axis-component.ts]: yep, done

File: projects/ngx-d3timeline/src/lib/options/options.selectors.ts (hunk 3 above)
Comment (wflemingnz-retired) [reply projects/ngx-d3timeline/src/lib/options/options.selectors.ts]: changed

**Review by umarmohammed — APPROVED** [review]

**Pull request merged**